api/src/main/java/org/openmrs/module/registrationcore/api/errorhandling/ErrorHandlingService.java

```diff
@@ -0,0 +1,6 @@
+package org.openmrs.module.registrationcore.api.errorhandling;
+
+public interface ErrorHandlingService {
+    
+    void handle(String messageBody, String destination, boolean failure, String failureReason);
```

Comment:
If this is an error handler, what's the purpose of the failure flag here? Since this is an error handler I would never expect it to be a success anyways.